astroid/bases.py

```diff
@@ -354,7 +353,16 @@ class Instance(BaseInstance):
     def __init__(self, proxied: nodes.ClassDef | None) -> None:
         super().__init__(proxied)
 
-    infer_binary_op: ClassVar[InferBinaryOp[Instance]]
+    @decorators.yes_if_nothing_inferred
+    def infer_binary_op(
+        self: Instance | nodes.ClassDef,
```

Comment:
This feels wrong, I don't think we should every type `self`. Why is this necessarry?

Reply:
merely copy paste. I'll clean it up to remove it.

Reply:
[c644b53](https://github.com/pylint-dev/astroid/pull/2171/commits/c644b53ea04af5e87cdf1ba536823dbe96b29775)